src/logic/contractInteraction/sources/ABIService/index.ts

```diff
@@ -32,10 +32,10 @@ export const getMethodSignatureAndSignatureHash = (
 
 export const extractUsefulMethods = (abi: AbiItem[]): AbiItemExtended[] => {
   return abi
-    .filter(({ constant, name, type }) => type === 'function' && !!name && typeof constant === 'boolean')
+    .filter(({ stateMutability, name, type }) => type === 'function' && !!name && stateMutability !== 'pure')
     .map(
       (method): AbiItemExtended => ({
-        action: method.constant ? 'read' : 'write',
+        action: method.stateMutability === 'view' ? 'read' : 'write',
```

Comment:
Why do we not allow to do a read call to `pure` functions?

Reply:
Mh... not sure.

Are there any functions we shouldn't list, apart from the `constructor` one?

Reply:
Not that I know of. What was the motivation to change this? Only the constructor case?

Reply:
The `constant` case was the issue here. It was assumed that the `constant` methods were the only ones we can interact with.

Reply:
I think you should turn around the logic, only "write" functions should be able to be used to trigger transactions. Also could we add some comments to this code so that everyone knows why we do this?

Reply:
Yes, I'll comment on this.

Just to make it clear, I should only use `function`s; marking `pure` and `view` as _read_ functions and all the others as _write_. Am I right?

Reply:
something around this?

```typescript
const isAllowedMethod = ({ name, type }: AbiItem): boolean => {
  return type === 'function' && !!name
}

const identifyMethodAction = ({ stateMutability }: AbiItem): 'read' | 'write' => {
  return ['view', 'pure'].includes(stateMutability) ? 'read' : 'write'
}

export const extractUsefulMethods = (abi: AbiItem[]): AbiItemExtended[] => {
  return abi
    .filter(isAllowedMethod)
    .map(
      (method): AbiItemExtended => ({
        action: identifyMethodAction(method),
        ...getMethodSignatureAndSignatureHash(method),
        ...method,
      }),
    )
    .sort(({ name: a }, { name: b }) => (a.toLowerCase() > b.toLowerCase() ? 1 : -1))
}
```

Reply:
@mikheevm :point_up_2:, WDYT?